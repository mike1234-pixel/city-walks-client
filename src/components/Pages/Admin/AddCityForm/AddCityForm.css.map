{"version":3,"sourceRoot":"","sources":["AddCityForm.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA","file":"AddCityForm.css"}