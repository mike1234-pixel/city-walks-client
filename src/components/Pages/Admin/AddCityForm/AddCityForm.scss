/** @format */

.add-city-form {
  padding-top: 20px;
  padding-bottom: 20px;
}
