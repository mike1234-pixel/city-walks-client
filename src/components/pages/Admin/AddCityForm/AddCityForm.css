.add-city-form {
  padding: 20px;
}
