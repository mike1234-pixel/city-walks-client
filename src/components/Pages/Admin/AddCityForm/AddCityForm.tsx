import React, { useState } from "react"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import axios, { AxiosError, AxiosResponse } from "axios"
import qs from "qs"
import toTitleCase from "../../../../functions/toTitleCase"
import City from "../../../../types/PostRequests/City"
import "./AddCityForm.css"

const AddCityForm: React.FC = () => {
  const [city, setCity] = useState<string>("")
  const [description, setDescription] = useState<string>("")
  const [img, setImg] = useState<string>("")

  const handleChange: (event: React.ChangeEvent<any>) => void = (event) => {
    switch (event.target.name) {
      case "city":
        setCity(event.target.value)
        break
      case "description":
        setDescription(event.target.value)
        break
      case "img":
        setImg(event.target.value)
        break
    }
  }

  const handleSubmit: (event: React.FormEvent) => void = (event) => {
    event.preventDefault()

    let payload: City = {
      city: toTitleCase(city),
      description: description,
      img: img,
    }

    axios
      .post("https://city-walks.herokuapp.com/add-city", qs.stringify(payload))
      .then((res: AxiosResponse) => {
        if (res.data === "city saved") {
          console.log("city saved")
        } else {
          console.log("city not saved")
        }
      })
      .catch((err: AxiosError) => {
        console.log(err)
      })

    alert("City Submitted")
    setCity("")
    setDescription("")
    setImg("")
    window.scrollTo(0, 0)
  }

  return (
    <div>
      <h2>Add City</h2>
      <form onSubmit={handleSubmit} className='add-city-form'>
        <MDBInput
          type='text'
          name='city'
          id='city'
          value={city}
          label='city'
          onChange={handleChange}
          maxLength='70'
          required
        />
        <MDBInput
          type='text'
          name='description'
          id='description'
          value={description}
          label='description'
          onChange={handleChange}
          maxLength='136'
          required
        />
        <MDBInput
          type='text'
          name='img'
          id='img'
          value={img}
          label='image link'
          onChange={handleChange}
          required
        />
        <MDBBtn outline color='elegant' type='submit'>
          Send City
          <MDBIcon far icon='paper-plane' className='ml-2' />
        </MDBBtn>
      </form>
    </div>
  )
}

export default AddCityForm
