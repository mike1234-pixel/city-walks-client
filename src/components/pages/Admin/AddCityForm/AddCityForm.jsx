import { useState } from "react"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import axios from "axios";
import qs from "qs";
import toTitleCase from '../../../../functions/toTitleCase'
import './AddCityForm.css'

const AddCityForm = () => {

    const [city, setCity] = useState("")
    const [description, setDescription] = useState("")
    const [img, setImg] = useState("")

    const handleChange = (event) => {
        switch(event.target.name) {
            case "city":
              setCity(event.target.value)
              break;
            case "description":
              setDescription(event.target.value)
              break;
            case "img":
              setImg(event.target.value)
              break;
          } 
    }

    const handleSubmit = (event) => {
        event.preventDefault()

        let payload = {
            city: toTitleCase(city),
            description: description,
            img: img
          };

        axios
        .post("http://localhost:5000/add-city", qs.stringify(payload))
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "city saved") {
            console.log("city saved")
          } else {
            console.log("city not saved")
          }
        });

        alert("City Submitted")
        setCity("")
        setDescription("")
        setImg("")
        window.scrollTo(0, 0);
    }

    return (
    <div>
        <h2>Add City</h2>
        <form onSubmit={handleSubmit} className="add-city-form">
            <MDBInput type="text" name="city" id="city" value={city} label="city" onChange={handleChange}  maxLength="70" required/>
            <MDBInput type="text" name="description" id="description" value={description} label="description" onChange={handleChange} maxLength="136" required/>
            <MDBInput type="text" name="img" id="img" value={img} label="image link" onChange={handleChange} required/>
            <MDBBtn outline color="elegant" type="submit">
                Send City
            <MDBIcon far icon="paper-plane" className="ml-2" />
          </MDBBtn>
        </form>
    </div>
    )
}

export default AddCityForm

// {
//     "id": "1",
//     "city": "london",
//     "description": "Capital of the UK",
//     "img_link": "https://images.pexels.com/photos/374815/pexels-photo-374815.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
// }