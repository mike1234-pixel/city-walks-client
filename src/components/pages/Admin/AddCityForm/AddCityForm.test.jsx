import renderer from 'react-test-renderer'
import AddCityForm from "./AddCityForm"

// snapshot test
it('AddCityForm Page renders correctly', () => {  

  const tree = renderer
    .create(
        <AddCityForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});