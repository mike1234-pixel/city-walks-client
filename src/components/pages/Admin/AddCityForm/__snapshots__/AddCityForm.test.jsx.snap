// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCityForm Page renders correctly 1`] = `
<div>
  <h2>
    Add City
  </h2>
  <form
    className="add-city-form"
    onSubmit={[Function]}
  >
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="city"
        maxLength="70"
        name="city"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="city"
        id=""
        onClick={[Function]}
      >
        city
      </label>
    </div>
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="description"
        maxLength="136"
        name="description"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="description"
        id=""
        onClick={[Function]}
      >
        description
      </label>
    </div>
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="img"
        name="img"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="img"
        id=""
        onClick={[Function]}
      >
        image link
      </label>
    </div>
    <button
      className="btn-elegant btn-outline-elegant btn Ripple-parent"
      data-test="button"
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Send City
      <i
        className="far fa-paper-plane ml-2"
        data-test="fa"
      />
      <div
        className="Ripple Ripple-outline "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </form>
</div>
`;
